### PR TITLE
Do shutdown from javascript

### PR DESCRIPTION
**Pull request by stefwalter**
Remove cockpitd code related to shutdown, and perform the shutdown from javascript.

### CODE REVIEW

**Comment by stefwalter** [comment]
@mvollmer It seems that at some point some shutdown workflows were lost. These can be added back later. This pull request is about straight refactoring, not adding back features.

**Comment by mvollmer** [comment]
We used to have an indication of a scheduled shutdown or restart and a cancel button for it when this feature had its own page.  This got removed when Shutdown/Restart was redone as a dialog.

File: pkg/shell/cockpit-server.js [hunk 1]
```diff
@@ -581,11 +571,15 @@ PageShutdownDialog.prototype = {
         else
             when = "+" + delay;
 
-        this.cockpitd_manager.call('Shutdown', op, when, message, function(error) {
-            $('#shutdown-dialog').modal('hide');
-            if (error && error.name != 'Disconnected')
-                shell.show_unexpected_error(error);
-        });
+        var arg = (op == "shutdown") ? "--poweroff" : "--reboot";
+        cockpit.spawn(["shutdown", arg, when, message])
```
Comment (mvollmer) [inline pkg/shell/cockpit-server.js]: { superuser: true }?

Comment (stefwalter) [inline pkg/shell/cockpit-server.js]: Hmmm, yes it seems that when specifying a time, root is required.

Comment (mvollmer) [inline pkg/shell/cockpit-server.js]: I also seem to need it for a immediate reboot... I'll check a bit more closely.

Comment (stefwalter) [inline pkg/shell/cockpit-server.js]: Perhaps it has to do with the message?

Comment (mvollmer) [inline pkg/shell/cockpit-server.js]: `shutdown --reboot now` asks for polkit auth and then reboots.

`shutdown --reboot +0` fails with "Must be root".  This is what we execute from Cockpit for "No Delay".

Still don't know why the test succeeds...

Comment (mvollmer) [inline pkg/shell/cockpit-server.js]: > Still don't know why the test succeeds...

Heh, it doesn't.  It waits for a reboot that never comes.  I thought I had seen a test run that failed in check-docker, but that was probably some other pull request.

Ok, it all makes sense again.  I'll add `superuser: true`, test, and merge.

**Pull request closed without merging**